pyEpiabm/pyEpiabm/sweep/host_progression_sweep.py

```diff
@@ -281,19 +283,117 @@ def __call__(self, time: float):
             Current simulation time
 
         """
+        asympt_uninf = []
         for cell in self._population.cells:
             for person in cell.persons:
                 if person.time_of_status_change is None:
                     assert person.infection_status \
                                     in [InfectionStatus.Susceptible]
+                    asympt_uninf.append((cell, person))
                     continue  # pragma: no cover
+                if person.infection_status in [InfectionStatus.Recovered,
+                                               InfectionStatus.Vaccinated]:
+                    asympt_uninf.append((cell, person))
                 while person.time_of_status_change <= time:
                     person.update_status(person.next_infection_status)
                     if person.infection_status in \
                             [InfectionStatus.InfectASympt,
                              InfectionStatus.InfectMild,
                              InfectionStatus.InfectGP]:
                         self.set_infectiousness(person, time)
+                        if not person.is_symptomatic():
+                            asympt_uninf.append((cell, person))
+                        self.sympt_testing_queue(cell, person)
                     self.update_next_infection_status(person)
                     self.update_time_status_change(person, time)
                 self._updates_infectiousness(person, time)
+
+        self.asympt_uninf_testing_queue(asympt_uninf, time)
+
+    def sympt_testing_queue(self, cell, person: Person):
+        """ Adds symptomatic people to a testing queue with a given
+        probability depedent on their status.
```

Comment:
This implies infection status whereas I think you mean their carehome/keyworker status